src/state/project-state.ts

```diff
@@ -1,53 +1,53 @@
-namespace App {
-  type Listener<T> = (items: T[]) => void;
-  class State<T> {
-    protected listeners: Listener<T>[] = [];
-    addListener(listenerFn: Listener<T>) {
-      this.listeners.push(listenerFn);
-    }
+import { Project } from "../models/project.js";
+import { ProjectStatus } from "../models/project.js";
+type Listener<T> = (items: T[]) => void;
+class State<T> {
+  protected listeners: Listener<T>[] = [];
+  //array of listeners where we store the functions that need to be called whenever the state changes or we add a project
+  addListener(listenerFn: Listener<T>) {
+    this.listeners.push(listenerFn);
+  }
+}
+export class ProjectState extends State<Project> {
+  private projects: Project[] = [];
+  //singleton we can only have one instance of project state class
+  private static instance: ProjectState;
```

Comment:
put the comment on top of the class , the comment is about the class ( how it is used) , not the static instance property ( it is hidden from outside the class)